sim/src/image.rs

```diff
@@ -139,11 +146,17 @@ impl ImagesBuilder {
     }
 
     /// Construct an `Images` that doesn't expect an upgrade to happen.
-    pub fn make_no_upgrade_image(self) -> Images {
+    pub fn make_no_upgrade_image(self, deps: &DepTest) -> Images {
         let mut flash = self.flash;
-        let images = self.slots.into_iter().map(|slots| {
-            let primaries = install_image(&mut flash, &slots[0], 42784, false);
-            let upgrades = install_image(&mut flash, &slots[1], 46928, false);
+        let num_images = self.slots.len();
```

Comment:
`let num_images = self.num_images();` ?

Reply:
Good catch. I added the `num_images` method after writing this, and didn't notice it. One tricky thing is that the call has to be moved above the `let mut flash = self.flash`, since that moves a value out of `self` making a method call invalid.